src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/eventDecoder.hpp

```diff
@@ -17,81 +17,150 @@
+                    data->cve5Buffer = builder.Release();
+                }
+                else
+                {
+                    data->translationsDatabase->put(data->resource.at("resource"), flatbufferResource);
+                }
+            }
 
-            jsonData.patch_inplace(data->resource.at("operations"));
-            if (!parser.Parse(jsonData.dump().c_str()))
+            else if ("update" == data->resource.at("type"))
             {
-                throw std::runtime_error("Unable to parse patched data: " + parser.error_);
+                rocksdb::PinnableSlice slice;
+                if (RESOURCE_TYPE::CVE5 == data->resourceType)
+                {
+                    if (!data->cvesDatabase->get(data->resource.at("resource"), slice))
+                    {
+                        throw std::runtime_error("Unable to find resource.");
+                    }
+                }
+                else
+                {
+                    if (!data->translationsDatabase->get(data->resource.at("resource"), slice))
+                    {
+                        throw std::runtime_error("Unable to find resource.");
+                    }
+                }
+
+                flatbuffers::IDLOptions options;
+                options.strict_json = true;
+                flatbuffers::Parser parser(options);
+                parser.Parse(schema);
+
+                std::string strData;
+                flatbuffers::GenText(parser, reinterpret_cast<const uint8_t*>(slice.data()), &strData);
+                auto jsonData = nlohmann::json::parse(strData);
+
+                jsonData.patch_inplace(data->resource.at("operations"));
+                if (!parser.Parse(jsonData.dump().c_str()))
+                {
+                    throw std::runtime_error("Unable to parse patched data: " + parser.error_);
+                }
+
+                rocksdb::Slice flatbufferResource(reinterpret_cast<const char*>(parser.builder_.GetBufferPointer()),
+                                                  parser.builder_.GetSize());
+                if (RESOURCE_TYPE::CVE5 == data->resourceType)
+                {
+                    data->cvesDatabase->put(data->resource.at("resource"), flatbufferResource);
+
+                    flatbuffers::FlatBufferBuilder& builder = parser.builder_;
+                    data->cve5Buffer = builder.Release();
+                }
+                else
+                {
+                    data->translationsDatabase->put(data->resource.at("resource"), flatbufferResource);
+                }
             }
 
-            rocksdb::Slice flatbufferResource(reinterpret_cast<const char*>(parser.builder_.GetBufferPointer()),
-                                              parser.builder_.GetSize());
-            data->cvesDatabase->put(data->resource.at("resource"), flatbufferResource);
+            else if ("delete" == data->resource.at("type"))
+            {
```

Comment:
I'd delete this block because the document removal isn't really supported, it'd be require to adapt many other things like the `storeModel`.

I did that here
https://github.com/wazuh/wazuh/pull/20529/files#diff-0835834f536f5e5a4bdea92a617a145ed260b4cf499d0410b0c4ee09f43762b0

Reply:
I kept the feature for translation with a comment to review the section later.